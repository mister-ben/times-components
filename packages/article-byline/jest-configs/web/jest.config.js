const path = require("path");

module.exports = {
  preset: "react-native",
  rootDir: "../../../..",
  transformIgnorePatterns: ["node_modules/(?!@times-components)/"],
  testMatch: [path.join(__dirname, "../../tests/**/*.test.web.js")],
  moduleNameMapper: {
    "react-native": "react-native-web"
  },
  moduleFileExtensions: ["web.js", "js", "json"],
  setupTestFrameworkScriptFile: `${__dirname}/setup-web-mocks.js`,
  testEnvironment: "jsdom"
};
