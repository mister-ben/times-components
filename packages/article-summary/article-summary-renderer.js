import React from "react";
import { Text } from "react-native";

export default {
  paragraph(key, attributes, renderedChildren) {
    return <Text key={key}>{renderedChildren} </Text>;
  },
  sentence(key, attributes, renderedChildren) {
    return <Text key={key}>{renderedChildren}. </Text>;
  },
  teaser(key, attributes, renderedChildren) {
    return <Text key={key}>{renderedChildren}...</Text>;
  }
};
