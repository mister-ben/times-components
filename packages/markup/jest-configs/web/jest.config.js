const path = require("path");

module.exports = {
  preset: "react-native",
  rootDir: "../../../..",
  transformIgnorePatterns: ["node_modules/(?!@times-components)/"],
  testMatch: [path.join(__dirname, "../../__tests__/**/*.test.web.js")],
  moduleNameMapper: {
    "react-native": "react-native-web"
  },
  moduleFileExtensions: ["web.js", "js", "json"],
  testEnvironment: "jsdom"
};
