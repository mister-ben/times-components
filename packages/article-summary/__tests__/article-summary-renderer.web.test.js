/* eslint-env jest */

import runTests from "./test-helper";
import { renderer } from "../article-summary-renderer.web";

describe("Article Summary Renderer Web", runTests(renderer));
