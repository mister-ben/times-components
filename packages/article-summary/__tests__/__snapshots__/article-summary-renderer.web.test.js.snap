// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Summary Renderer Web renders a paragraph ending with a full stop and a space with no new line 1`] = `
<span>
  Content
  . 
</span>
`;

exports[`Article Summary Renderer Web renders a teaser with an ellipsis at the end 1`] = `
<span>
  Content
  ...
</span>
`;
